homeassistant/components/lyric/sensor.py

```diff
@@ -0,0 +1,289 @@
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        lyric: Lyric,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            lyric,
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_outdoor_temperature",
+            "Outdoor Temperature",
+            "mdi:thermometer",
+            hass.config.units.temperature_unit,
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        for location in self.coordinator.data:
+            for device in location.devices:
+                if device.macID == self._device.macID:
+                    return device.outdoorTemperature
+
+
+class LyricOutdoorHumiditySensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        lyric: Lyric,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            lyric,
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_outdoor_humidity",
+            "Outdoor Humidity",
+            "mdi:water-percent",
+            "%",
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        for location in self.coordinator.data:
+            for device in location.devices:
+                if device.macID == self._device.macID:
+                    return device.displayedOutdoorHumidity
+
+
+class LyricNextPeriodSensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        lyric: Lyric,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            lyric,
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_next_period_time",
+            "Next Period Time",
+            "mdi:clock",
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        for location in self.coordinator.data:
+            for device in location.devices:
+                if device.macID == self._device.macID:
+                    time = dt_util.parse_time(device.changeableValues.nextPeriodTime)
+                    now = dt_util.utcnow()
+                    if time <= now.time():
+                        now = now + timedelta(days=1)
+                    return dt_util.as_utc(datetime.combine(now.date(), time))
```

Comment:
This doesn't display correctly. Is this the correct format?

![Screenshot_20200915-123052.jpg](https://user-images.githubusercontent.com/28114703/93205273-74117680-f74f-11ea-8e05-b45c418d8aed.jpg)

Reply:
Device classes. Fixed